aws/templates/solution/solution_ElasticSearch.ftl

```diff
@@ -1,125 +1,129 @@
 [#-- ElasticSearch --]
-[#if (componentType == "elasticsearch" ||
-        (componentType == "es")) && deploymentSubsetRequired("es", true)]
-    [#assign es = component.ES!component.ElasticSearch]
-    [#assign esId = formatElasticSearchId(
-                        tier,
-                        component)]
+[#if (componentType == ES_COMPONENT_TYPE || componentType == "elasticsearch" || componentType == "es") ]
 
-    [#assign processorProfile = getProcessor(tier, component, "ElasticSearch")]
-    [#assign master = processorProfile.Master!{}]
+    [#list requiredOccurrences(
+        getOccurrences(tier, component),
+        deploymentUnit) as occurrence]
+        
+        [@cfDebug listMode occurrence false /]
 
-    [#assign storageProfile = getStorage(tier, component, "ElasticSearch")]
-    [#assign volume = (storageProfile.Volumes["codeontap"])!{}]
+        [#assign core = occurrence.Core]
+        [#assign configuration = occurrence.Configuration]
+        [#assign resources = occurrence.State.Resources]
 
-    [#assign esCIDRs =
-                getUsageCIDRs(
-                    "es",
-                    es.IPAddressGroups![])]
-    [#list zones as zone]
-        [#assign zoneIP =
-            getExistingReference(
-                formatComponentEIPId("mgmt", "nat", zone),
-                IP_ADDRESS_ATTRIBUTE_TYPE
-            )
-        ]
-        [#if zoneIP?has_content]
-            [#assign esCIDRs += [zoneIP] ]
-        [/#if]
-    [/#list]
-    [#list 1..20 as i]
-        [#assign externalIP =
-            getExistingReference(
-                formatComponentEIPId("mgmt", "nat", "external" + i)
-            )
-        ]
-        [#if externalIP?has_content]
-            [#assign esCIDRs += [externalIP] ]
-        [/#if]
-    [/#list]
+        [#assign esId = resources["es"].Id]
 
-    [#assign esAdvancedOptions = {} ]
-    [#list es.AdvancedOptions as option]
-        [#assign esAdvancedOptions +=
-            {
-                option.Id : option.Value
-            }
-        ]
-    [/#list]
+        [#assign processorProfile = getProcessor(tier, component, "ElasticSearch")]
+        [#assign master = processorProfile.Master!{}]
 
-    [#-- In order to permit updates to the security policy, don't name the domain. --]
-    [#-- Use tags in the console to find the right one --]
-    [#-- "DomainName" : "${productName}-${segmentId}-${tierId}-${componentId}", --]
+        [#assign storageProfile = getStorage(tier, component, "ElasticSearch")]
+        [#assign volume = (storageProfile.Volumes["codeontap"])!{}]
+        [#assign esCIDRs =
+                    getUsageCIDRs(
+                        "es",
+                        configuration.IPAddressGroups)]
+        [#list zones as zone]
```

Comment:
This code should be converted to use links to the IP component in other products. Links would make it much cleaner